### PR TITLE
adjoint for cumsum

### PR DESCRIPTION
**Pull request by mcabbott**
The easy half of #282

### CODE REVIEW

File: src/lib/array.jl [hunk 1]
```diff
@@ -191,6 +191,13 @@ end
 _backmean(xs, Δ, ::Colon) = zero(xs) .+ Δ ./ length(xs)
 _backmean(xs, Δ, dims) = zero(xs) .+ Δ ./ mapreduce(i -> size(xs,i),*,dims)
 
+@adjoint function cumsum(xs::AbstractArray; dims = 1)
```
Comment (CarloLucibello) [inline src/lib/array.jl]: `dims` should have no default value, according to Base signature

**Comment by CarloLucibello** [comment]
@mcabbott can you rebase this?

**Comment by mcabbott** [comment]
Sure, will revise. I think this may have had some complications to avoid missing methods of `reverse(::CuArray)` which may be fixed now, will check.

**Comment by mcabbott** [comment]
I just pushed changes to my branch, but for some strange reason can't see them here. Am I going mad or is something broken? https://github.com/FluxML/Zygote.jl/compare/master...mcabbott:cumsum

**Comment by mcabbott** [comment]
Should be ready now.

**Comment by dhairyagandhi96** [comment]
CI seems to be stuck

**Comment by CarloLucibello** [comment]
just checked that `reverse(::CuArray; dims) works fine with CuArrays.allowscalar(false). Merging this, thanks!

bors r+

**Pull request merged**